### PR TITLE
convert importance rank pp to use the new data objects

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------
##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by wangcj05** [comment]
I think this is ready for you to review. @alfoa @PaulTalbot-INL

File: framework/PostProcessors/ImportanceRank.py [hunk 1]
```diff
@@ -295,126 +225,17 @@ def collectOutput(self,finishedJob, output):
...
-        output.setExtension(outputExtensions)
-      output.setPath(self._workingDir)
-      self.raiseADebug('Dumping output in file named ' + output.getAbsFile())
-      output.open('w')
-      if outputExtension == 'csv':
-        self._writeCSV(output,outputDict)
-      else:
-        self._writeXML(output,outputDict)
+    print("Debug Realization: ")
+    print(outputDict)
```
Comment (PaulTalbot-INL) [inline framework/PostProcessors/ImportanceRank.py]: Debug statements.

File: framework/PostProcessors/ImportanceRank.py [hunk 2]
```diff
@@ -454,69 +275,64 @@ def inputToInternal(self, currentInp):
...
-        for feat in self.features:
-          if feat in currentInput.getParaKeys('input'):
-            inputDict['timeDepData'][pivotParameter[step]]['features'][feat] = np.asarray([inputs[i][feat][-1] for i in inputs.keys()])
-          else:
-            self.raiseAnError(IOError, self, 'Feature ' + feat + ' has not been found in data object '+currentInput.name)
-        inputDict['timeDepData'][pivotParameter[step]]['metadata'] = currentInput.getAllMetadata()
-
+      self.pivotValue = dataSet[self.pivotParameter].values
+      if not currentInput.checkIndexAlignment(indexesToCheck=self.pivotParameter):
+        self.raiseAnError(IOError, "The data provided by the data objects ", currentInput.name, " is not synchronized!")
```
Comment (PaulTalbot-INL) [inline framework/PostProcessors/ImportanceRank.py]: This message might be expanded to clarify what "synchronized" means; i.e., that the realizations' pivot parameters have unsynchronized pivot values, and that the HistorySetSync internal PP can fix it.

File: framework/PostProcessors/ImportanceRank.py [hunk 3]
```diff
@@ -454,69 +275,64 @@ def inputToInternal(self, currentInp):
...
+      slices = currentInput.sliceByIndex(self.pivotParameter)
+      metadata = currentInput.getMeta(pointwise=True)
+      inputList = []
+      for sliceData in slices:
+        inputDict = {}
+        inputDict['metadata'] = metadata
```
Comment (PaulTalbot-INL) [inline framework/PostProcessors/ImportanceRank.py]: This metadata is being re-set every iteration?

File: framework/PostProcessors/ImportanceRank.py [hunk 4]
```diff
@@ -545,26 +361,22 @@ def __runLocal(self, inputDict):
...
       if what.lower() == 'sensitivityindex':
         what = 'sensitivityIndex'
-        if what not in outputDict.keys():
-          outputDict[what] = {}
         for target in self.targets:
-          entries = senWeightDict[target]
-          entries.sort(key=lambda x: x[1],reverse=True)
-          outputDict[what][target] = entries
+          for i, feat in enumerate(feats):
+            varName = what + '_' + target + '_' + feat
```
Comment (PaulTalbot-INL) [inline framework/PostProcessors/ImportanceRank.py]: No need to change, but less chance for error if this is constructed as `'_'.join([what,target,feat])`.  This happens a couple times below, as well.

File: framework/PostProcessors/ImportanceRank.py (hunk 1 above)
Comment (wangcj05) [reply framework/PostProcessors/ImportanceRank.py]: removed

File: framework/PostProcessors/ImportanceRank.py (hunk 2 above)
Comment (wangcj05) [reply framework/PostProcessors/ImportanceRank.py]: updated.

File: framework/PostProcessors/ImportanceRank.py (hunk 4 above)
Comment (wangcj05) [reply framework/PostProcessors/ImportanceRank.py]: fixed.

File: framework/PostProcessors/ImportanceRank.py (hunk 3 above)
Comment (wangcj05) [reply framework/PostProcessors/ImportanceRank.py]: Yes, this is a quick fix. We can change it if we start to clean up the inputToInternal.

**Pull request merged**